src/morphology.cpp

```diff
@@ -129,7 +129,7 @@ const std::vector<Section> Morphology::sections() const
 {
     // TODO: Make this more performant when needed
     std::vector<Section> sections_;
```

Comment:
What about using `std::vector::reserve` before filling `sections_`?
It would also use `emplace_back` to avoid the copies (including the one in `section` member method).

```
sections_.emplace_back(id, _properties);
```

Reply:
Good catch!

Reply:
not sure there is a move ctr for Section, something to look into/add